drivers/serial/uart_gecko.c

```diff
@@ -303,451 +303,136 @@ static const struct uart_driver_api uart_gecko_driver_api = {
 
 #define DT_DRV_COMPAT silabs_gecko_uart
 
-#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
-
-#define PIN_UART0_RXD {DT_INST_PROP_BY_IDX(0, location_rx, 1), \
-		DT_INST_PROP_BY_IDX(0, location_rx, 2), gpioModeInput, 1}
-#define PIN_UART0_TXD {DT_INST_PROP_BY_IDX(0, location_tx, 1), \
-		DT_INST_PROP_BY_IDX(0, location_tx, 2), gpioModePushPull, 1}
-
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
-static void uart_gecko_config_func_0(struct device *dev);
-#endif
-
-static const struct uart_gecko_config uart_gecko_0_config = {
-	.base = (USART_TypeDef *)DT_INST_REG_ADDR(0),
-	.clock = CLOCK_UART(DT_INST_PROP(0, peripheral_id)),
-	.baud_rate = DT_INST_PROP(0, current_speed),
-	.pin_rx = PIN_UART0_RXD,
-	.pin_tx = PIN_UART0_TXD,
-#ifdef CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION
-	.loc_rx = DT_INST_PROP_BY_IDX(0, location_rx, 0),
-	.loc_tx = DT_INST_PROP_BY_IDX(0, location_tx, 0),
+#define GECKO_UART_IRQ_HANDLER_DECL(index)				       \
```

Comment:
There doesn't seem to be a convention, even in case of gecko drivers, however (IMO) using a shorter name for `index` reads better. In the long form it reads as if it was some kind of property. In case of gecko family in gpio_gecko.c we have `idx`, in wdt_gecko.c `index` and in spi_gecko.c `n`. Other drivers use mostly `idx` or `n`.

Reply:
replaced index with idx.